internal/scanner/scanner.go

```diff
@@ -52,13 +54,57 @@ type Scanner struct {
 	printJSON  bool
 }
 
-// NewScanner creates a new scanner with the PoliciesFetcher provided
-func NewScanner(policiesFetcher PoliciesFetcher, resourcesFetcher ResourcesFetcher, printJSON bool) (*Scanner, error) {
+// NewScanner creates a new scanner with the PoliciesFetcher provided. If
+// insecureClient is false, it will read the caCertFile and add it to the in-app
+// cert trust store. This gets used by the httpCLient when connection to
+// PolicyServers endpoints.
+func NewScanner(policiesFetcher PoliciesFetcher, resourcesFetcher ResourcesFetcher,
+	printJSON bool,
+	insecureClient bool, caCertFile string) (*Scanner, error) {
 	report, err := report.NewPolicyReportStore()
 	if err != nil {
 		return nil, err
 	}
-	return &Scanner{policiesFetcher, resourcesFetcher, *report, http.Client{}, printJSON}, nil
+
+	// configure client for TLS
+	var httpClient http.Client
+	if insecureClient {
+		tr := &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}} //nolint
+		httpClient = http.Client{Transport: tr}
+		log.Info().Msg("connecting to PolicyServers endpoints in insecure SSL mode")
+	} else {
```

Comment:
Because of this `else` statement, the following scenario is not handled: `--insecure-ssl --filepath-cert foo`. So either we raise and error when both flags are used at the same time, or we change the code slightly.

Reply:
indeed, redone it in https://github.com/kubewarden/audit-scanner/pull/75/commits/7049c72aaf34f2a973670baf8b39e3c185e8db4c, to allow for `--insecure` and `--extra-ca` at the same time.